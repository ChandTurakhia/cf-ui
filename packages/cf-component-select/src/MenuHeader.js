import React from 'react';
import PropTypes from 'prop-types';
import ReactSelect from 'react-select';
import { createComponent } from 'cf-style-container';

const MenuHeader = createComponent(
  ({ theme }) => ({
    color: theme.color.dusk,
    cursor: 'default',
    fontWeight: 'bold',
    fontSize: '.86667rem',
    backgroundColor: theme.color.moonshine,
    borderBottom: `1px solid ${theme.color.dust}`,
    width: '100%',
    '> *': {
      display: 'inline-block',
      padding: '8px 10px'
    }
  }),
  'h3'
);

export default MenuHeader;
