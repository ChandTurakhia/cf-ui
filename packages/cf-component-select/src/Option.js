import React from 'react';
import ReactSelect from 'react-select';
import { createComponent } from 'cf-style-container';

const Option = createComponent(({
  theme,
  isSelected,
  isFocused,
  isDisabled
}) => {
  const color = isSelected || isFocused
    ? theme.color.charcoal
    : theme.colorGray;

  let backgroundColor = theme.colorWhite;
  if (isFocused) {
    backgroundColor = '#f0f0f0';
  } else if (isSelected) {
    backgroundColor = '#f7f7f7';
  }

  return {
    boxSizing: 'border-box',
    cursor: 'pointer',
    display: 'block',
    padding: '8px 10px',
    opacity: isDisabled ? 0.5 : 1,
    backgroundColor,
    color,
    '& [role="option"]': {
      whiteSpace: 'nowrap',
      overflow: 'hidden',
      textOverflow: 'ellipsis'
    }
  };
}, ReactSelect.defaultProps.optionComponent);

export default Option;
