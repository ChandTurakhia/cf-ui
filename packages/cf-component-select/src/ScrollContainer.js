import React from 'react';
import PropTypes from 'prop-types';
import { createComponent } from 'cf-style-container';

class ScrollContainer extends React.Component {
  constructor(props) {
    super(props);
    this.handleMouseOver = this.handleMouseOver.bind(this);
    this.handleMouseOut = this.handleMouseOut.bind(this);
  }

  // Prevent body scrolling when the menu scrolls
  handleMouseOver() {
    if (this.ref.scrollHeight > this.ref.clientHeight) {
      document.body.style.overflow = 'hidden';
    }
  }

  handleMouseOut() {
    if (this.ref.scrollHeight > this.ref.clientHeight) {
      document.body.style.overflow = 'initial';
    }
  }

  componentWillUnmount() {
    document.body.style.overflow = 'initial';
  }

  render() {
    return (
      <div
        className={this.props.className}
        onScroll={this.handleScroll}
        onMouseOver={this.handleMouseOver}
        onMouseOut={this.handleMouseOut}
        ref={ref => this.ref = ref}
      >
        {this.props.children}
      </div>
    );
  }
}

ScrollContainer.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string
};

const ScrollContainerWrapper = createComponent(theme => {
  return {
    maxHeight: theme.maxMenuHeight || '198px',
    overflowY: 'auto'
  };
}, ScrollContainer);

export default ScrollContainerWrapper;
