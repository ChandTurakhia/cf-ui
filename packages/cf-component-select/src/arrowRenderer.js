import React from 'react';
import PropTypes from 'react';

import { createComponent } from 'cf-style-container';

const ArrowRendererWrapper = createComponent(({ theme, isOpen }) => ({
  borderColor: isOpen
    ? `transparent transparent ${theme.color.cement}`
    : `${theme.color.cement} transparent transparent`,
  borderWidth: isOpen ? `0 5px 5px` : '5px 5px 2.5px',
  borderStyle: 'solid',
  display: 'inline-block',
  height: 0,
  width: 0,
  position: 'relative',
  top: isOpen ? '-2px' : 0
}));

export default function arrowRenderer({ onMouseDown }) {
  return <ArrowRendererWrapper onMouseDown={onMouseDown} />;
}

arrowRenderer.propTypes = {
  onMouseDown: PropTypes.func
};
