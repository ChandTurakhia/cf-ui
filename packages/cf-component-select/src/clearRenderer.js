import React from 'react';
import { createComponent } from 'cf-style-container';

const ClearRenderer = createComponent(
  ({ theme }) => ({
    display: 'inline-block',
    fontSize: '18px',
    fontWeight: 'normal',
    lineHeight: 1,
    ':hover': {
      color: theme.color.red
    }
  }),
  'span'
);

export default function clearRenderer() {
  return (
    <ClearRenderer>
      <span dangerouslySetInnerHTML={{ __html: '&times;' }} />
    </ClearRenderer>
  );
}
