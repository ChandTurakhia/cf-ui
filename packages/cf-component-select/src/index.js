import { applyTheme } from 'cf-style-container';

import SelectUnstyled from './Select';
import SelectTheme from './SelectTheme';

const Select = applyTheme(SelectUnstyled, SelectTheme);

export { Select, SelectUnstyled, SelectTheme };
