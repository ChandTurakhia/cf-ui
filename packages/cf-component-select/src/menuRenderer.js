import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { createComponent } from 'cf-style-container';
import ScrollContainer from './ScrollContainer';

const MenuWrapper = createComponent(({ theme }) => {
  return {
    border: `1px solid ${theme.color.hail}`,
    borderRadius: theme.borderRadius,
    backgroundColor: theme.colorWhite,
    marginTop: '5px',

    boxSizing: 'border-box',
    position: 'absolute',
    top: '100%',
    minWidth: '100%',
    maxWidth: '350px',
    zIndex: '1100',
    webkitOverflowScrolling: 'touch'
  };
});

export default function menuRenderer(
  {
    focusedOption,
    instancePrefix,
    labelKey,
    onFocus,
    onSelect,
    optionClassName,
    optionComponent,
    optionRenderer,
    options,
    valueArray,
    valueKey,
    onOptionRef
  },
  { menuTitle, closeMenu, closeMenuComponent, menuHeaderComponent }
) {
  const MenuHeader = menuHeaderComponent;
  const CloseMenu = closeMenuComponent;
  const Option = optionComponent;

  const opts = options.map((option, i) => {
    let isSelected = valueArray && valueArray.indexOf(option) > -1;
    let isFocused = option === focusedOption;

    return (
      <Option
        instancePrefix={instancePrefix}
        isDisabled={option.disabled}
        isFocused={isFocused}
        isSelected={isSelected}
        onFocus={onFocus}
        onSelect={onSelect}
        option={option}
        optionIndex={i}
        ref={ref => {
          onOptionRef(ref, isFocused);
        }}
        key={`option-${i}-${option[valueKey]}`}
      >
        {optionRenderer(option, i)}
      </Option>
    );
  });

  return (
    <MenuWrapper>
      {menuTitle &&
        <MenuHeader>
          <span>{menuTitle}</span>
          <CloseMenu closeMenu={closeMenu} />
        </MenuHeader>}
      <ScrollContainer>
        {opts}
      </ScrollContainer>
    </MenuWrapper>
  );
}

menuRenderer.propTypes = {
  focusedOption: PropTypes.object,
  instancePrefix: PropTypes.string,
  labelKey: PropTypes.string,
  onFocus: PropTypes.func,
  onSelect: PropTypes.func,
  optionClassName: PropTypes.string,
  optionComponent: PropTypes.element,
  optionRenderer: PropTypes.func,
  options: PropTypes.array,
  valueArray: PropTypes.array,
  valueKey: PropTypes.string,
  onOptionRef: PropTypes.func
};
