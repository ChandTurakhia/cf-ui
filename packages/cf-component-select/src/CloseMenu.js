import React from 'react';
import PropTypes from 'prop-types';
import { createComponent } from 'cf-style-container';
import clearRenderer from './clearRenderer';

class CloseMenuComponent extends React.Component {
  constructor(props) {
    super(props);
    this.handleTouchDown = () => this.dragging = true;
    this.handleTouchMove = () => this.dragging = false;
    this.handleTouchEnd = () => {
      if (!this.dragging) {
        this.props.closeMenu();
      }
    };
  }
  render() {
    return (
      <span
        className={this.props.className}
        onMouseDown={this.props.closeMenu}
        onTouchStart={this.handleTouchStart}
        onTouchMove={this.handleTouchMove}
        onTouchEnd={this.handleTouchEnd}
      >
        {clearRenderer()}
      </span>
    );
  }
}

CloseMenuComponent.propTypes = {
  closeMenu: PropTypes.func.isRequired,
  className: PropTypes.string
};

const CloseMenu = createComponent(
  () => ({
    cursor: 'pointer',
    float: 'right'
  }),
  CloseMenuComponent
);

export default CloseMenu;
