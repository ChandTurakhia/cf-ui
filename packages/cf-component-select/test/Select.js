import React from 'react';
import { felaSnapshot } from 'cf-style-provider';
import { Select } from '../../cf-component-select/src/index';

test('should render', () => {
  const snapshot = felaSnapshot(<Select />);
  expect(snapshot.component).toMatchSnapshot();
  expect(snapshot.styles).toMatchSnapshot();
});
