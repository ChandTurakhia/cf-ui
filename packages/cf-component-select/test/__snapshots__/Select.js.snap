// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="a b c d e f g h i j k l m n o p q r s t u v w x y z ab ac ae af ag ah ai aj ak al am an ao ap aq ar as at au av aw ax ay az ba bb bc bd be bf bg bh bi bj bk bl bm bn bo bp bq br bs bt bu bv bw bx by bz ca cb cc cd ce cf cg ch ci cj ck cl cm cn co cp cq cr cs ct cu cv cw cx cy cz"
>
  <div
    className="Select Select--single"
    style={undefined}
  >
    <div
      className="Select-control"
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <span
        className="Select-multi-value-wrapper"
        id="react-select-2--value"
      >
        <div
          className="Select-placeholder"
        >
          Select...
        </div>
        <div
          aria-activedescendant="react-select-2--value"
          aria-expanded={false}
          aria-owns="react-select-2--value"
          aria-readonly="false"
          className="Select-input"
          onBlur={[Function]}
          onFocus={[Function]}
          role="combobox"
          style={
            Object {
              "border": 0,
              "display": "inline-block",
              "width": 1,
            }
          }
          tabIndex={0}
        />
      </span>
      <span
        className="Select-arrow-zone"
        onMouseDown={[Function]}
      >
        <div
          className="da db dc dd de df n dg"
        />
      </span>
    </div>
  </div>
</div>
`;

exports[`should render 2`] = `
"
.a {
  box-sizing: border-box
}

.b {
  font-size: 0.86667rem
}

.c div {
  box-sizing: border-box
}

.d div {
  font-size: 0.86667rem
}

.e input {
  box-sizing: border-box
}

.f input {
  font-size: 0.86667rem
}

.g span {
  box-sizing: border-box
}

.h span {
  font-size: 0.86667rem
}

.i {
  flex: initial;
  -webkit-flex: initial
}

.j {
  margin: initial
}

.k {
  width: initial
}

.l {
  display: initial
}

.m {
  vertical-align: initial
}

.n {
  position: relative
}

.o {
  opacity: 1
}

.p .Select-control {
  background-color: #fff
}

.q .Select-control {
  border: 1px solid #BCBEC0
}

.r .Select-control {
  border-radius: 2px
}

.s .Select-control {
  color: #333333
}

.t .Select-control {
  cursor: default
}

.u .Select-control {
  display: table
}

.v .Select-control {
  border-spacing: 0px
}

.w .Select-control {
  border-collapse: separate
}

.x .Select-control {
  height: 32px
}

.y .Select-control {
  outline: none
}

.z .Select-control {
  overflow: hidden
}

.ab .Select-control {
  position: relative
}

.ac .Select-control {
  width: 100%
}

.ae .Select-control {
  transition: border-color 0.2s ease;
  -webkit-transition: border-color 0.2s ease;
  -moz-transition: border-color 0.2s ease
}

.af .Select-control .Select-input:focus {
  outline: none
}

.ag .Select-menu-outer {
  position: relative
}

.ah:hover .Select-control {
  border-color: #2869a2
}

.ai .is-focused .Select-control {
  border-color: #2869a2
}

.aj .is-open .Select-control {
  border-color: #BCBEC0
}

.ak .Select-placeholder {
  bottom: 0px
}

.al .Select-placeholder {
  color: #333
}

.am .Select-placeholder {
  left: 0px
}

.an .Select-placeholder {
  line-height: 32px
}

.ao .Select-placeholder {
  padding-left: 10px
}

.ap .Select-placeholder {
  padding-right: 10px
}

.aq .Select-placeholder {
  position: absolute
}

.ar .Select-placeholder {
  right: 0px
}

.as .Select-placeholder {
  top: 0px
}

.at .Select-placeholder {
  max-width: 100%
}

.au .Select-placeholder {
  overflow: hidden
}

.av .Select-placeholder {
  text-overflow: ellipsis
}

.aw .Select-placeholder {
  white-space: nowrap
}

.ax .Select--single > .Select-control .Select-value {
  bottom: 0px
}

.ay .Select--single > .Select-control .Select-value {
  color: #333
}

.az .Select--single > .Select-control .Select-value {
  left: 0px
}

.ba .Select--single > .Select-control .Select-value {
  line-height: 32px
}

.bb .Select--single > .Select-control .Select-value {
  padding-left: 10px
}

.bc .Select--single > .Select-control .Select-value {
  padding-right: 10px
}

.bd .Select--single > .Select-control .Select-value {
  position: absolute
}

.be .Select--single > .Select-control .Select-value {
  right: 0px
}

.bf .Select--single > .Select-control .Select-value {
  top: 0px
}

.bg .Select--single > .Select-control .Select-value {
  max-width: 100%
}

.bh .Select--single > .Select-control .Select-value {
  overflow: hidden
}

.bi .Select--single > .Select-control .Select-value {
  text-overflow: ellipsis
}

.bj .Select--single > .Select-control .Select-value {
  white-space: nowrap
}

.bk .Select-input {
  height: 32px
}

.bl .Select-input {
  padding-left: 10px
}

.bm .Select-input {
  padding-right: 10px
}

.bn .Select-input {
  vertical-align: middle
}

.bo .Select-input > input {
  width: 100%
}

.bp .Select-input > input {
  background: none transparent
}

.bq .Select-input > input {
  border: 0 none
}

.br .Select-input > input {
  cursor: default
}

.bs .Select-input > input {
  display: inline-block
}

.bt .Select-input > input {
  font-family: inherit
}

.bu .Select-input > input {
  font-size: inherit
}

.bv .Select-input > input {
  margin: 0px
}

.bw .Select-input > input {
  outline: none
}

.bx .Select-input > input {
  line-height: 14px
}

.by .Select-input > input {
  padding: 0px
}

.bz .Select-noresults {
  box-vizing: border-box
}

.ca .Select-noresults {
  color: #7D7D7D
}

.cb .Select-noresults {
  cursor: default
}

.cc .Select-noresults {
  display: block
}

.cd .Select-noresults {
  padding: 8px 10px
}

.ce .Select-aria-only {
  display: inline-block
}

.cf .Select-aria-only {
  height: 1px
}

.cg .Select-aria-only {
  width: 1px
}

.ch .Select-aria-only {
  margin: -1px
}

.ci .Select-aria-only {
  clip: rect(0, 0, 0, 0)
}

.cj .Select-aria-only {
  overflow: hidden
}

.ck .Select-arrow-zone {
  cursor: pointer
}

.cl .Select-arrow-zone {
  display: table-cell
}

.cm .Select-arrow-zone {
  position: relative
}

.cn .Select-arrow-zone {
  text-align: center
}

.co .Select-arrow-zone {
  vertical-align: middle
}

.cp .Select-arrow-zone {
  width: 25px
}

.cq .Select-arrow-zone {
  padding-right: 5px
}

.cr .Select-arrow-zone {
  top: 1px
}

.cs .Select-clear-zone {
  color: #7D7D7D
}

.ct .Select-clear-zone {
  cursor: pointer
}

.cu .Select-clear-zone {
  display: table-cell
}

.cv .Select-clear-zone {
  position: relative
}

.cw .Select-clear-zone {
  top: 1px
}

.cx .Select-clear-zone {
  text-align: center
}

.cy .Select-clear-zone {
  vertical-align: middle
}

.cz .Select-clear-zone {
  width: 17px
}

.da {
  border-color: #7D7D7D transparent transparent
}

.db {
  border-width: 5px 5px 2.5px
}

.dc {
  border-style: solid
}

.dd {
  display: inline-block
}

.de {
  height: 0px
}

.df {
  width: 0px
}

.dg {
  top: 0px
}
"
`;
